index.js

```diff
@@ -1,5 +1,78 @@
 import { AppRegistry } from 'react-native';
 import App from './src/';
 import { name as appName } from './app.json';
+import { isValidHttpUrl } from './src/libraries';
+import { FloatingModule } from './src/nativeModules/get';
 
+function getNoticeIds(matchingContexts, eventMessageFromChromeURL) {
+  return matchingContexts
+    .map((res) => {
+      const addWWWForBuildingURL = `www.${eventMessageFromChromeURL}`;
+
+      if (addWWWForBuildingURL.match(new RegExp(res.urlRegex, 'g'))) {
+        return res.noticeId;
+      }
+    })
+    .filter(Boolean);
+}
+
+const HeadlessTask = async (taskData) => {
+  const eventMessageFromChromeURL = taskData.url;
+  if (eventMessageFromChromeURL && isValidHttpUrl(eventMessageFromChromeURL)) {
+    console.log('FETCH MATCHING CONTEXT');
+    const matchingContexts = await fetch(
```

Comment:
All matching contexts and notices (domain?) related code should probably live in dedicated file or folder.